sdk/core/System.ClientModel/src/Options/LoggingOptions.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+
+namespace System.ClientModel.Options;
+
+/// <summary>
+/// TODO
+/// </summary>
+public class LoggingOptions
+{
+    private bool _frozen;
+
+    private const int DefaultLoggedContentSizeLimit = 4 * 1024;
+    private const bool DefaultIsLoggingEnabled = true;
+    private const bool DefaultIsLoggingContentEnabled = false;
+    private static readonly IList<string> s_defaultLoggedHeaderNames =
```

Comment:
TODO: Do we want to have some defaults for the headers and queries or just sanitize everything by default? I took these from Azure Core and removed the Azure/Microsoft related headers